src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -523,7 +524,7 @@ int ControllerScriptInterfaceLegacy::beginTimer(
     m_timers[timerId] = info;
     if (timerId == 0) {
         m_pScriptEngineLegacy->logOrThrowError(QStringLiteral("Script timer could not be created"));
-    } else if (oneShot) {
+    } else if (oneShot && CmdlineArgs::Instance().getControllerDebug()) {
         qCDebug(m_logger) << "Starting one-shot timer:" << timerId;
     } else {
         qCDebug(m_logger) << "Starting timer:" << timerId;
```

Comment:
this seems half backed now, either check it for both or none. Though shouldn't it be unnecessary to check, because the logging category takes care of it?

Reply:
Ah indeed, needs more doing.
By category, I assume you mean the debug category? If yes, this is unfortunately not enough about the noisy logging related to controller communication is blindly fed to the debug category everywhere on the code base.

Reply:
ah, yeah, thats unfortunate. I think we have some code somewhere though that adjusts the default logging category. That should get rid of the controller debug code in theory. Maybe thats broken? Can you look into it?

Reply:
Fair enough, I didn't realise this was supposed to be a feature since #4523

I'm not sure if it doesn't work as expected or if I misunderstood the point of having a dedicated CLI arg - are we expecting controller debug message to show when running with `--log-level debug`?

Reply:
> I'm not sure if it doesn't work as expected or if I misunderstood the point of having a dedicated CLI arg 

Well, the dedicated CLI flag was there first, so we tried our best to keep it for backwards compatibility. 

> are we expecting controller debug message to show when running with --log-level debug?

I'm actually not sure. I'd have to go through the PR again to look it up. Either way, it seems broken, right?

Reply:
> Either way, it seems broken, right?

My assumption from reading the `--help` output was that it wouldn't log controller related debug message if not explicitly opted-in so with that understanding, yes it's broken. 
But we could also say the `man` is misleading and that `--controller-debug` will enable debug message only for controller, whereas `--log-level debug` enable **all** log messages, including controller ones.

IMO, it should be `1)` - when I implemented the S4 Mk3 mapping, I add to add that [nasty workaround](https://github.com/mixxxdj/mixxx/blob/42a1fb8ba0da9e83a50bbeaabc9042e5b2b59098/src/controllers/hid/hidioglobaloutputreportfifo.cpp#L81), otherwise I would get a 10G+ logfile after an hour session, due to that message logging every single HID packet coming in every 20ms

Reply:
@Swiftb0yare you happy to keep this work around as-is and create a follow up issue + `FIXME` comment?

Reply:
works for me, yes